content/blog/2019/egress-traffic-control-in-istio-part-3/index.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Secure Control of Egress Traffic in Istio, part 3
+subtitle: Comparison with alternative solutions for control of egress and performance considerations
+description: Comparison with alternative solutions for control of egress traffic and performance considerations.
+publishdate: 2019-07-16
+attribution: Vadim Eisenberg (IBM)
+keywords: [traffic-management,egress,security,gateway,tls]
+---
+
+Welcome to part 3 in our new series about secure control of egress traffic in Istio.
+In [the first part in the series](/blog/2019/egress-traffic-control-in-istio-part-1/), I presented the attacks involving
+egress traffic and the requirements we collected for a secure control system for egress traffic.
+In [the second part in the series](/blog/2019/egress-traffic-control-in-istio-part-2/), I presented the Istio way of
+securing egress traffic and showed how you can prevent the attacks using Istio.
+
+In this installment, I compare secure control of egress traffic in Istio with alternative solutions such as Kubernetes
+Network Policies and legacy egress proxies/firewalls. Finally, I describe the performance considerations regarding the
+secure control of egress traffic in Istio.
+
+## Alternative solutions for egress traffic control
+
+First, let me remind you the [requirements for egress traffic control we collected](/blog/2019/egress-traffic-control-in-istio-part-1/#requirements-for-egress-traffic-control):
+
+1.  Support for [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) with
+    [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) or for [TLS origination](/docs/reference/glossary/#tls-origination) by Istio.
+1.  **Monitor** SNI and the source workload of every egress access.
+1.  Define and enforce **policies per cluster**.
+1.  Define and enforce **policies per source**, _Kubernetes-aware_.
+1.  **Prevent tampering**.
+1.  Traffic control is **transparent** to the applications.
+
+Now, let me present two alternative solutions for egress traffic control and show which requirements they satisfy, and,
+more importantly, cannot satisfy.
+
+The most natural solution for egress traffic control is
```

Comment:
@rcaballeromx Kubernetes Network Policies is the "native" Kubernetes solution. For a person familiar with Kubernetes, this would be the first mechanism to come to mind in the context of traffic control. However, it does not fulfill all the requirements. I am not trying to "sell Istio", and as part of my blog post I present the alternative solutions. This section is an answer to the questions I encountered when presenting egress traffic control in Istio: how is it better than Kubernetes Network Policies? why should I use Istio for egress traffic control and not a DNS-aware egress proxy?

Reply:
Addressed by https://github.com/istio/istio.io/pull/4637/commits/1dbe203d93535b322fca096db0a2a5a20991ca3b.